src/module/item/item.js

```diff
@@ -93,6 +93,20 @@ export class ItemArchmage extends Item {
     await this._handleBreathSpell(itemToRender);
     await this._handleRetainFocus(itemToRender, hitEvalRes, actorUpdateData, chatData);
 
+    // Set a flag for stoke adjustments.
+    if (game.settings.get('archmage', 'secondEdition')) {
+      if (this.actor.type === 'npc' && this.actor.system.details.type.value === 'dragon') {
```

Comment:
As above, this looks like it should just check for the stoke resource being active.